### DISCUSSION TITLE
OpenTelemetry and FluidFramework

### DISCUSSION THREAD

**Discussion by jvoels** [microsoft/FluidFramework#7106] [Ideas]
I wanted to open a pretty generic discussion on this topic based on commentary in this [pull request](https://github.com/microsoft/FluidFramework/pull/6838#discussion_r676991077). It seems like we identified opportunity to review any potential for discovering and applying, if any, valuable [OpenTelemetry ](https://opentelemetry.io/) concepts to our framework code that is instrumented for logging. Since Fluid is likely to be a key part of many cloud-native distributed product stories the logging capabilities provided by the framework will likely be a part of many distributed tracing challenges.

[OpenTelemetry](https://opentelemetry.io/docs/concepts/what-is-opentelemetry/) presents the case it can add value through standardization in this space. And Microsoft is [not a stranger](https://cloudblogs.microsoft.com/opensource/2019/05/23/announcing-opentelemetry-cncf-merged-opencensus-opentracing/) to the potential interactions here.

I know Fluid and FRS have been doing work on defining interfaces for logging from fluid pieces and refining instrumentation. Tagging some folks for this discussion + awareness as we invest in these spaces.

@markfields @vladsud @andre4i @hedasilv @TriDucNgo @znewton @jatgarg

**Comment by hedasilv** [comment]
On the server side, we have plans to explore OpenTelemetry as soon as it is mature. According to [this page](https://opentelemetry.io/docs/js/), OpenTelemetry for JS is still in Alpha for metrics, and beta for tracing. There is no mention about logs. Also, I engaged with people at Microsoft working on OpenTelemetry who confirmed, in June, that "OpenTelemetry JavaScript SDK for Tracing is not in Stable release state (while most other languages are). Overall Metrics for OpenTelemetry is in the works and could get to Stable by EOY".
